pipeline-runner/tests/testthat/test-qc.R

```diff
@@ -50,13 +50,32 @@ clean_timestamp <- function(scdata) {
   return(scdata)
 }
 
+# Seurat object commands are sometimes functions which will have a new environment
+# for every call which breaks snapshots
+remove_commands_functions <- function(data) {
+  command_names <-  names(data@commands)
+  for (command_name in command_names) {
+    param_names <- names(data@commands[[command_name]]@params)
+
+    for (param_name in param_names) {
+      param <- data@commands[[command_name]]@params[[param_name]]
+
+      if (methods::is(param, 'function'))
+        data@commands[[command_name]]@params[[param_name]] <- NULL
+    }
+  }
+  return(data)
+}
```

Comment:
you could move these copies to a [helper file](https://testthat.r-lib.org/articles/special-files.html#helper-files).

Reply:
I remember having read somewhere that it was discouraged, and instead suggested to put these things in the R/ dir so that they can also be tested, but it's not in the vignette now...